compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -91,12 +203,39 @@ trait ConstraintHandling {
       // so we shouldn't allow them as constraints either.
       false
     else
-      val dropWildcards = new AvoidWildcardsMap:
-        if !isUpper then variance = -1
-        override def mapWild(t: WildcardType) =
-          if approximateWildcards then super.mapWild(t)
-          else newTypeVar(apply(t.effectiveBounds).toBounds)
-      val bound = dropWildcards(rawBound)
+      val bound =
+        if this.isInstanceOf[GadtConstraint] then
```

Comment:
This is the only place where we refer to GadtConstraint. I believe it would be cleaner to break out bound computation into a method such as `avoidingBound` that is overridden in GadtConstraint.